design/apply-config-from-cli.md

```diff
@@ -0,0 +1,68 @@
+# Allow applying application configuration from CLI as a stand-alone operation
+
+Admin Console CLI has a way to specify appication configuration during the initial install.
+However there is no dedicated way to use CLI to update application configuration.
+`kots download` and `kots upload` commands can be used to achieve this goal.
+However, this does not allow to easily patch the config without knowing the current set of keys and values.
+
+## Goals
+
+- Configuration can be applied independently from the rest of the app spec (unlike kots download/upload commands).
+- Configuration can be applied to new versions of the app before they are deployed.
+- Individual config keys can be added, modified, and removed.
+
+## Non Goals
+
+- Changing current architecture, in which config changes are always separate from other changes, such as license updates or application version updates.
+License updates and upstream updates will still create their own versions with existing config.
+- Covering every add/delete/modify config key cases in convenience commands, though the CLI can be extended later.
+- Interractive editing in terminal.
+- Adding the ability to make config changes to a deployed application without having to also take the latest downloaded upstream app version. 
+
+## Background
+
+Some application require periodic configuration updates.
+CLI will provide a way to automate such operations.
+
+## High-Level Design
+
+There will be two modes of operation:
+1. Apply an entire configuration to the app.
+1. Apply a config change to a subset of values.  This is mostly a convenience method since this can be achieved with the first mode.
+
+## Detailed Design
+
+The following command will be added to CLI:
+
+```
+kubectl kots config <app-slug> set [parameters]
+```
+
+The following flags will be supported, in addition to all higher level flags:
+
+```
+ --config-values <path to config file>
+ --merge
+ --config-key
+ --config-value
+ --config-value-from-file
+ --deploy
+ ```
+
+ | Flag | Description |
+| :---- | ----------- |
+| `--config-values` | This is a path to a config file compiant with KOT's `kind: ConfigValues` format.  By default, the contents of this file will be used to replace existing app config in its entirety.  This property can be used to delete config items that are no longer used. |
```

Comment:
`--config-values` is confusable with `--config-value`. How about `--config-file` instead? Or even just `--config`, especially if we use `--set` for individual config?

Reply:
Feature creep, but it would be nice to support a URL for values as well, especially with `--merge` as a vendor could provide some sample configuration, while allowing people to leave other config alone.

Reply:
how about `--from-file`?

Reply:
Hmm... interesting. Since the command is `set config`, I guess the `config` part of the arguments is redundant. Are you suggesting `kots set config --from-file config.yaml` to load the whole thing, or is `--from-file` for setting an individual value? 

Maybe put it all together and see how it works. What's it look like if we support:
1. key=value
2. individual values from file
3. whole config from file
4. k8s-style verb-then-noun